sources.pkr.hcl

```diff
@@ -7,7 +7,7 @@ source "amazon-ebs" "base" {
   secret_key      = var.aws_secret_access_key
 
   ami_name      = "${local.image_name}-${var.architecture}-${local.now_unix_timestamp}"
-  instance_type = local.aws_spot_instance_types[var.architecture]
+  instance_type = [local.aws_spot_instance_types[var.architecture][0]]
```

Comment:
it need to be a string !
https://developer.hashicorp.com/packer/integrations/hashicorp/amazon/latest/components/builder/ebs#run-configuration